docs/resources/azure_generic_resource.md

```diff
@@ -1,27 +1,80 @@
+* `attribute` - This generic resource dynamically creates the attributes on the fly based on the type of resource that has been targetted.
+* `value` is the expected output from the chosen attribute
+
+The options that can be passed to the resource are as follows.
+
+| Name        | Description                                                                                                         | Required | Example                           |
+|-------------|---------------------------------------------------------------------------------------------------------------------|----------|-----------------------------------|
+| group_name: | Azure Resource Group to be tested                                                                                   | yes      | MyResourceGroup                   |
+| name:       | Name of the Azure resource to test                                                                                  | no       | MyVM                              |
+| type: | Azure Resource Type to look for | no | Microsoft.Compute/virtualMachines |
+| apiversion: | API Version to use when interrogating the resource. If not set then the latest version for the resoure type is used | no       | 2017-10-9                         |
 
-As an alternative the following environment variables can be set instead of passing the options to the control:
+These options can also be set using the environment variables:
 
  - `AZURE_RESOURCE_GROUP_NAME`
  - `AZURE_RESOURCE_NAME`
  - `AZURE_RESOURCE_TYPE`
  - `AZURE_RESOURCE_API_VERSION`
 
-Note that if options are set on the control and in enviroment variables, the environment variables take perference.
+When the options have been set as well as the environment variables, the environment variables take priority.
 
 There are _normally_ three standard tests that can be performed on a resource.
 
 | Name | Description |
 |------|-------------|
 | name | Name of the resource |
 | type | Type of resource |
-| location | Location of the resource within Azure |
\ No newline at end of file
+| location | Location of the resource within Azure |
+
+For example:
+
+```ruby
+describe azure_generic_resource(group_name: 'Inspec-Azure', name: 'Linux-Internal-VM') do
+  its('location') { should eq 'westeurope' }
+end
+```
+
+## Attributes
+
+The attributes that can be tested are entirely dependent on the Azure Resource that is under scrutiny. That means the attributes vary. The best way to see what is available please use the [Azure Resources Portal](https://resources.azure.com) to select the resource you are interested in and see what can be tested.
+
+![Virtual Machine Attributes](images/linux_internal_vm_resource.png)
+
+This resource allows you to test _any_ valid Azure Resource. The trade off for this is that the language to check each item is not as natural as it would be for a native Inspec resource.
+
+Please see the integration tests for in depth examples of how this resource can be used.
```

Comment:
There are a lot of integration tests with lots of comments in them. I do not see the value of recreating these in Markdown when they are explicilty linked to in the documenation.
I agree that if this just told people to look in the Integration Test folder then this would be a problem, but as there are hyperlinks I do not think it would be an issue.